### PR TITLE
network_base: add check_firewall_backend function

### PR DESCRIPTION
**Pull request by yanqzhan**
Add a function to judge nftables or iptables to use for test.

### CODE REVIEW

**Comment by yanqzhan** [comment]
Tested in https://github.com/autotest/tp-libvirt/pull/5739.

File: provider/virtual_network/network_base.py [hunk 1]
```diff
@@ -336,3 +336,23 @@ def check_iface_attrs(iface, key, expect_val):
...
+    host_distro = distro.detect().name
+    host_version = distro.detect().version
+    from_rhel10 = (host_distro == 'rhel' and int(host_version) >= 10)
+    if from_rhel10:
+        firewall_backend = "nftables"
+        with open('/etc/libvirt/network.conf') as file:
+            for line in file.readlines():
+                if re.search('^firewall_backend\\s*=\\s*.iptables', line):
+                    firewall_backend = "iptables"
+                    break
```
Comment (Yingshun) [inline provider/virtual_network/network_base.py:356]: Can you plz reuse utils_config in avocado-vt? If there is no proper class, can you add a one for libvirt network config?

File: provider/virtual_network/network_base.py [hunk 2]
```diff
@@ -336,3 +336,23 @@ def check_iface_attrs(iface, key, expect_val):
...
+
+def check_firewall_backend():
+    """
+    Check host firewall_backend to determine to use nft or iptables cmd
+
+    """
+    firewall_backend = "iptables"
+    host_distro = distro.detect().name
+    host_version = distro.detect().version
+    from_rhel10 = (host_distro == 'rhel' and int(host_version) >= 10)
```
Comment (Yingshun) [inline provider/virtual_network/network_base.py:349]: I wonder if there is another way to check the firewall backend or not. can we check what the default firewalld on host is?

Comment (chloerh) [reply provider/virtual_network/network_base.py:349]: +1

Comment (yanqzhan) [reply provider/virtual_network/network_base.py:349]: I'm not sure, after discussion with some colleagues, the methods maybe:
1.check comments in /etc/libvirt/network.conf：
#   If firewall_backend isn't configured, libvirt will choose the
#   first available backend from the following list:
#
#     [iptables, nftables]
2. check *.conf.in, which's the source file of .conf file. But I didn't find network.conf.in, and they're in source code.
3. check logs while executing corresponding actions.

File: provider/virtual_network/network_base.py (hunk 1 above)
Comment (yanqzhan) [reply provider/virtual_network/network_base.py:356]: I can do it later. Thx.

File: provider/virtual_network/network_base.py [hunk 3]
```diff
@@ -336,3 +337,29 @@ def check_iface_attrs(iface, key, expect_val):
...
+        with open('/etc/libvirt/network.conf') as file:
+            for line in file.readlines():
+                if re.search('^firewall_backend\\s*=\\s*.iptables', line):
+                    firewall_backend = "iptables"
+                    break
+    elif libvirt_version.version_compare(10, 4, 0):
+        with open('/etc/libvirt/network.conf') as file:
+            for line in file.readlines():
+                if re.search('^firewall_backend\\s*=\\s*.nftables', line):
+                    LOG.debug(f'my values: {firewall_backend}')
```
Comment (dzhengfy) [inline provider/virtual_network/network_base.py:361]: It seems this debugging message is not useful enough.
IF you would like to keep it, please avoid oral words , like my,  and give a meaningful sentence.

File: provider/virtual_network/network_base.py (hunk 1 above)
Comment (dzhengfy) [reply provider/virtual_network/network_base.py:356]: just execute like
```
`grep  ^#firewall_backend\\s*=\\s*.nftables /etc/libvirt/network.conf`
#firewall_backend = "nftables"
```

Could it meet your needs?

**Comment by yanqzhan** [comment]
After discussion, reviewer thinks this is not a good method. And from another discussion with feature owner and auto owner, we decide not to check rules anymore but only cares about funciton in other cases. So close it.

**Pull request closed without merging**